pantab/__init__.py

```diff
@@ -14,3 +14,67 @@
     "frames_to_hyper",
     "test",
 ]
+
+# We link against HyperAPI in a fun way: In Python, we extract the function
+# pointers directly from the Python HyperAPI. We pass those function pointers
+# over to the C module which will then use those pointers to directly interact
+# with HyperAPI. Furthermore, we check the function signatures to guard
+# against API-breaking changes in HyperAPI.
+#
+# Directly using HyperAPI's C functions always was and still is discouraged and
+# unsupported by Tableu. In particular, Tableau will not be able to provide
+# official support for this hack.
+#
+# Because this is highly brittle, we try to make the error message as
+# actionable as possible and guide users in the right direction.
+
+api_incompatibility_msg = """
+pantab is incompatible with the version of Tableau Hyper API installed on your
+system. Please upgrade both `tableauhyperapi` and `pantab` to the latest version.
+If doing so does not fix this issue, please file an issue at
+https://github.com/innobi/pantab/issues mentioning the exact pantab and HyperAPI
+versions which triggered this error.
+"""
+
+try:
+    from tableauhyperapi.impl.dll import lib, ffi
+except ImportError as e:
+    raise NotImplementedError(api_incompatibility_msg) from e
+
+
+def _get_hapi_function(name, sig):
+    assert hasattr(lib, name), f"function '{name}' missing"
+    f = getattr(lib, name)
+    func_type = ffi.typeof(f)
+    assert (
```

Comment:
I know this will seem pedantic but can you explicitly raise on these rather than using assert? Some users do use the "-O" flag and they would be losing out on these messages. 

The first one here is definitely a TypeError, I'm thinking the second error is ValueError but if you see something more fitting go for it

Reply:
good point.

I changed the checks to directly raise a `NotImplementedError` instead of some other subclass. In the end, we would have wrapped the more specific sub-exceptions into a `NotImplementedError` anyway